travis/getaddons.py

```diff
@@ -102,6 +104,57 @@ def get_modules_changed(path, ref='HEAD'):
     return modules_changed_path
 
 
+def get_dependencies(modules, module_name):
+    result = set()
+    for dependency in modules.get(module_name, {}).get('depends', []):
+        result |= get_dependencies(modules, dependency)
+    return result | set([module_name])
+
+
+def get_dependents(modules, module_name):
+    result = set()
+    for dependent in modules.keys():
+        if module_name in modules.get(dependent, {}).get('depends', []):
+            result |= get_dependents(modules, dependent)
+    return result | set([module_name])
+
+
+def add_auto_install(modules, to_install):
+    """ Append automatically installed glue modules to to_install if their
+    dependencies are already present. to_install is a set. """
+    found = True
+    while found:
+        found = False
+        for module in modules.keys():
```

Comment:
<s>Remove `.keys()`</s>
<s>Use `for module in modules` instead.</s>

In fact, you are using the key to get value then you could use:
`for module, module_data in dicc.items():`